packages/shared/src/components/post/SquadPostAuthor.tsx

```diff
@@ -42,7 +42,7 @@ function SquadPostAuthor({
             <span className={classNames('font-bold', className?.name)}>
               {author.name}
             </span>
-            <SquadMemberBadge key="squadMemberRole" role={role} />
+            {role && <SquadMemberBadge key="squadMemberRole" role={role} />}
```

Comment:
Better to use `!!role` 

Typescript does guard us from this code wise but this is good in general because even though react does not render `false` or `''` to the DOM if for example any non truthy attribute is provided eg. `0` it will actually render `0` to DOM which breaks UI.